cmd/ctr/import.go

```diff
@@ -6,31 +6,62 @@ import (
 	"os"
 
 	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/contrib/docker1"
+	"github.com/containerd/containerd/imageformats"
+	oci "github.com/containerd/containerd/imageformats/oci"
 	"github.com/containerd/containerd/log"
 	"github.com/urfave/cli"
 )
 
 var imagesImportCommand = cli.Command{
-	Name:        "import",
-	Usage:       "import an image",
-	ArgsUsage:   "[flags] <ref> <in>",
-	Description: `Import an image from a tar stream.`,
-	Flags: []cli.Flag{
+	Name:      "import",
+	Usage:     "import an image",
+	ArgsUsage: "[flags] <in>",
+	Description: `Import an image from a tar stream.
+Implemented formats:
+- oci.v1     (default)
+- docker.v1
+`,
+	Flags: append([]cli.Flag{
 		cli.StringFlag{
-			Name:  "ref-object",
-			Value: "",
-			Usage: "reference object e.g. tag@digest (default: use the object specified in ref)",
+			Name:  "format",
+			Value: "oci.v1",
+			Usage: "image format. See DESCRIPTION.",
+		},
+		cli.StringFlag{
+			Name:  "docker.v1-prefix",
+			Value: "imported/docker.v1/",
+			Usage: "prefix added to docker.v1 RepoTags",
+		},
+		cli.StringFlag{
+			Name:  "oci.v1-prefix",
+			Value: "imported/oci.v1/imported:",
+			Usage: "prefix added to oci.v1 ref annotation",
 		},
 		labelFlag,
-	},
+	}, snapshotterFlags...),
+
 	Action: func(clicontext *cli.Context) error {
 		var (
-			ref       = clicontext.Args().First()
-			in        = clicontext.Args().Get(1)
-			refObject = clicontext.String("ref-object")
-			labels    = labelArgs(clicontext.StringSlice("label"))
+			in            = clicontext.Args().First()
+			labels        = labelArgs(clicontext.StringSlice("label"))
+			imageImporter imageformats.Importer
 		)
 
+		switch format := clicontext.String("format"); format {
+		case "oci.v1":
+			// TODO(AkihiroSuda): how to import OCI images without ref annotation?
```

Comment:
Just use the image digest as the name.

Reply:
OK, but anyway name(ref) spec needs some "locator" and "object" atm

Can we relax the spec?